quickget

```diff
@@ -3326,6 +3286,72 @@ function create_vm() {
     make_vm_config "${ISO}"
 }
 
+function create_config() {
+    local VM_PATH="${1}"
+    local INPUT="${2}"
+    local FIXED_ISO=""
+
+    OS="custom"
+    if ! mkdir "${VM_PATH}" 2>/dev/null; then
+        echo "ERROR! Could not create directory: ${VM_PATH}. Please verify that it does not already exist"
+        exit 1
+    fi
+    if [[ "${INPUT}" == "http://"* ]] || [[ "${INPUT}" == "https://"* ]]; then
+        INPUT="$(web_redirect "${INPUT}")"
+        if [[ "${INPUT}" == *".iso" ]] || [[ "${INPUT}" == *".img" ]]; then
+            web_get "${INPUT}" "${VM_PATH}"
+            INPUT="${VM_PATH}/${INPUT##*/}"
+        else
+            echo "ERROR! Only ISO,IMG and QCOW2 file types are supported for --create-config"
+            exit 1
+        fi
+    fi
+    if [[ "${INPUT}" == *".iso" ]]; then
+        echo "Moving image to VM dir" && mv "${INPUT}" "${VM_PATH}"
+        CUSTOM_IMAGE_TYPE="iso"
+    elif [[ "${INPUT}" == *".img" ]]; then
+        echo "Moving image to VM dir" && mv "${INPUT}" "${VM_PATH}"
+        CUSTOM_IMAGE_TYPE="img"
+    elif [[ "${INPUT}" == *".qcow2" ]]; then
+        echo "Moving image to VM dir" && mv "${INPUT}" "${VM_PATH}/disk.qcow2"
+        CUSTOM_IMAGE_TYPE="qcow2"
+    else
+        echo "ERROR! Only ISO,IMG and QCOW2 file types are supported for --create-config"
+        exit 1
+    fi
+    echo "Creating custom VM config for ${INPUT##*/}."
+    case "${INPUT,,}" in
+        *windows-server*) CUSTOM_OS="windows-server";;
+        *windows*)
+            CUSTOM_OS="windows"
+            while [ $# -gt 2 ]; do
+                case "${3}" in
+                    -unattended|--unattended)
+                        echo "Downloading VirtIO drivers..."
+                        web_get "https://fedorapeople.org/groups/virt/virtio-win/direct-downloads/stable-virtio/virtio-win.iso" "${VM_PATH}"
+                        FIXED_ISO=${VM_PATH}/virtio-win.iso
+
+                        rm -f "${VM_PATH}/unattended.iso"
+                        unattended_windows "${VM_PATH}"
+                        shift
+                esac
+            done
+            ;;
+
+        *freebsd*) CUSTOM_OS="freebsd";;
+        *kolibrios*) CUSTOM_OS="kolibrios";;
+        *reactos*) CUSTOM_OS="reactos";;
+        *) CUSTOM_OS="linux";;
+    esac
+    echo "Selecting OS: ${CUSTOM_OS}. If this is incorrect, please modify the config file to include the correct OS."
+    echo
+    if [ -n "${FIXED_ISO}" ]; then
+        make_vm_config "${INPUT}" "${FIXED_ISO}"
+    else
+        make_vm_config "${INPUT}"
+    fi
```

Comment:
The check for whether the fixed ISO parameter has a non-empty value is already present within make_vm_config. Remove the unnecessary check.
```suggestion
    make_vm_config "${INPUT}" "${FIXED_ISO}"
```